### ISSUE TITLE
Cache locations and Mac App Store rejection.

### ISSUE THREAD

**Issue by theofanis** [akavache/Akavache#228]
Our Xamarin.Mac app got rejected because it requests read/write in `~/.local`, since `BlobCache.LocalMachine` is created under this folder. Same about `~/.config` and the other two caches.

The report contained the text I'm pasting below:

```
The application accesses the following location(s): '~/.local' 

The majority of developers encountering this issue are opening files in Read/Write mode instead of Read-Only mode, in which case it should be changed to Read-Only. 

Other common reasons for this issue include: 
- creating or writing files in the above location(s), which are not valid locations for files to be written as stated in documentation 
- writing to the above location(s) without using a valid app-id as a container for the written files 

Please review the "File-System Usage Requirements for the App Store" of Submitting to the Mac App Store for the locations apps are allowed to write and for further guidance.
```

According to https://github.com/akavache/Akavache/blob/master/Akavache/MacFilesystemProvider.cs the locations should be:

`NSSearchPathDirectory.CachesDirectory` (Library/Caches)
`NSSearchPathDirectory.ApplicationSupportDirectory` (Library/Application Support)

but our app creates them under
`~/.local/share/<appname>`
`~/.config/<appname>`

We have a PCL referencing Akavache for our app caching needs, which targets the standard pcl platforms (.NET 4.5, Xamarin.iOS etc) and our GUI app is a Xamarin.Mac project targeting .NET 4.5 (not the Unified API). Are we maybe missing something?

/cc @paulcbetts

**Comment by theofanis** [comment]
OK, we figured it out. The problem was that when the Xamarin.Mac app targeting .NET 4.5 referenced Akavache, nuget added the `.net45` version reference.

Akavache correctly used `SimpleFileSystemProvider` and not `MacFileSystemProvider`, which created the files under the `System.Environment.SpecialFolders.ApplicationData` and `System.Environment.SpecialFolders.LocalApplicationData`, which return the paths I mentioned above.

[Xamarin.Mac docs](http://macapi.xamarin.com/?link=T%3aSystem.Environment%2bSpecialFolder) indeed inform us about these paths.

We manually changed the referenced Akavache.dll to use the Xamarin.Mac10 version, and everything was a smooth sail from there on. The Xamarin.Mac app runs without any issues creating (and accessing) the caches in the correct "valid" locations. 

Debug runs indeed create the caches under `Library/Caches/<appname>` and `Library/Application Support/<appname>` and AppStore sandbox release builds use the same paths but under `Library/Containers/<app-identifier>/`, where `<app-identifier>` is our bundle's identifier (`com.company.product` pattern). So the final cache locations for the AppStore sandbox build are:

```
Library/Containers/<app-identifier>/Library/Caches/<appname>/BlobCache
Library/Containers/<app-identifier>/Library/Application Support/<appname>/BlobCache
Library/Containers/<app-identifier>/Library/Application Support/<appname>/SecretCache
```

Sorry for the hassle. Here's hoping this information helps others as well.
